src/renderer/entities/wallet/lib/permission-utils.ts

```diff
@@ -0,0 +1,143 @@
+import { Account, ProxiedAccount, Wallet } from '@shared/core';
+import { walletUtils } from './wallet-utils';
+
+export const permissionUtils = {
+  isTransferAvailable,
+  isReceiveAvailable,
+  isStakingAvailable,
+  isCreateMultisigTxAvailable,
+  isApproveMultisigTxAvailable,
+  isRejectMultisigTxAvailable,
+  isCreateAnyProxyAvailable,
+  isCreateNonAnyProxyAvailable,
+  isRemoveProxyAvailable,
+};
+
+function isTransferAvailable(wallet: Wallet, accounts: Account[]): boolean {
+  if (walletUtils.isWatchOnly(wallet)) {
+    return false;
+  }
+
+  if (walletUtils.isProxied(wallet)) {
```

Comment:
Maybe additionally add `accountUtils.isProxiedAccount(accounts[0])`
to exclude `accounts[0] as ProxiedAccount` but that's not exactly true 😊